tests/rptest/tests/rbac_test.py

```diff
@@ -49,70 +51,80 @@ def setUp(self):
+                                                 self.role_name1))
 
         with expect_role_error(RoleErrorCode.ROLE_NOT_FOUND):
-            self.superuser_admin.update_role_members(role=role_name1,
+            self.superuser_admin.update_role_members(role=self.role_name1,
                                                      add=[ALICE.username])
 
         res = self.superuser_admin.list_roles(filter='ba',
                                               principal=ALICE.username)
         assert len(RolesList.from_response(res)) == 0, "Unexpected roles"
 
         with expect_role_error(RoleErrorCode.ROLE_NOT_FOUND):
-            self.superuser_admin.list_role_members(role=role_name1)
+            self.superuser_admin.list_role_members(role=self.role_name1)
 
         res = self.superuser_admin.list_user_roles()
         assert len(RolesList.from_response(res)) == 0, "Unexpected user roles"
 
         with expect_role_error(RoleErrorCode.ROLE_NOT_FOUND):
-            self.superuser_admin.delete_role(role=role_name1)
+            self.superuser_admin.delete_role(role=self.role_name1)
 
     @cluster(num_nodes=3)
     def test_regular_user_access(self):
         # a regular user may NOT access the RBAC API
 
-        role_name0 = 'foo'
-        role_name1 = 'bar'
-
         with expect_http_error(403):
             self.user_admin.list_roles()
 
         with expect_http_error(403):
-            self.user_admin.create_role(role=role_name0)
+            self.user_admin.create_role(role=self.role_name0)
 
         with expect_http_error(403):
-            self.user_admin.get_role(role=role_name0)
+            self.user_admin.get_role(role=self.role_name0)
 
         with expect_http_error(403):
-            self.user_admin.update_role(role=role_name0,
-                                        update=RoleUpdate(role_name1))
+            self.user_admin.update_role(role=self.role_name0,
+                                        update=RoleUpdate(self.role_name1))
 
         with expect_http_error(403):
-            self.user_admin.update_role_members(role=role_name1,
+            self.user_admin.update_role_members(role=self.role_name1,
                                                 add=[ALICE.username])
 
         with expect_http_error(403):
-            self.user_admin.list_role_members(role=role_name1)
+            self.user_admin.list_role_members(role=self.role_name1)
 
         res = self.user_admin.list_user_roles()
         assert len(
             RolesList.from_response(res)) == 0, "Unexpected roles for user"
 
         with expect_http_error(403):
-            self.user_admin.delete_role(role=role_name1)
+            self.user_admin.delete_role(role=self.role_name1)
+
+    @cluster(num_nodes=3)
+    def test_create_role(self):
+        res = self.superuser_admin.create_role(role=self.role_name0)
+        created_role = res.json()['role']
+        assert created_role == self.role_name0, f"Incorrect create role response: {res.json()}"
```

Comment:
Meant to comment before - I think we'd be well served to test misuse of the API here, specifically the malformed body cases. Also idempotency.

Reply:
Good call. Because I used `parse_json_body`, we return a 400 with a non-role-specific error if the body is missing / not well-formed json. I think that's fine and not worth the trouble changing (RPK/console should just send us well-formed requests anyway). But maybe we should call this out in the RFC. What do you think?

Reply:
SGTM 👍